src/scala/com/github/johnynek/bazel_deps/CoursierResolver.scala

```diff
@@ -51,57 +51,96 @@ class CoursierResolver(servers: List[MavenServer], ec: ExecutionContext, runTime
-          case (None, a) => downloadSha1(a)
+          case (None, a) => downloadSha(digestType, a)
         }
 
-      def downloadSha1(a: coursier.Artifact): Task[Option[ResolvedSha1Value]] =
+      def downloadSha(digestType: DigestType, a: coursier.Artifact): Task[Option[ShaValue]] =
         Cache.file[Task](a).run.map {
           case Left(error) =>
             logger.info(s"failure to download ${a.url}, ${error.describe}")
             None
           case Right(file) =>
             val serverId = serverFor(a).fold("")(_.id)
-            val o = Sha1Value.parseFile(file).map(s => ResolvedSha1Value(s, serverId)).toOption
+            val o = ShaValue.parseFile(digestType, file).toOption
             o.foreach { r =>
-              logger.info(s"SHA-1 for ${c.asString} downloaded from ${a.url} (${r.sha1Value.toHex})")
+              logger.info(s"SHA-1 for ${c.asString} downloaded from ${a.url} (${r.toHex})")
             }
             o
         }
 
-      def computeSha1(artifact: coursier.Artifact): Task[ResolvedSha1Value] =
+      def computeSha(digestType: DigestType, artifact: coursier.Artifact): Task[ShaValue] =
         Cache.file[Task](artifact).run.flatMap { e =>
           resolverMonad.fromTry(e match {
             case Left(error) =>
               Failure(FileErrorException(error))
             case Right(file) =>
-              Sha1Value.computeShaOf(file).map { sha =>
-                logger.info(s"SHA-1 for ${c.asString} computed from ${artifact.url} (${sha.toHex})")
-                val serverId = serverFor(artifact).fold("")(_.id)
-                ResolvedSha1Value(sha, serverId)
-              }
+              ShaValue.computeShaOf(digestType, file)
           })
         }
 
-      def computeSha1s(as: List[coursier.Artifact]): Task[ResolvedSha1Value] =
+      def computeShas(digestType: DigestType, as: List[coursier.Artifact], errorMessage: Option[String] = None): Task[ShaValue] =
         as match {
-          case Nil => resolverMonad.raiseError(new RuntimeException(s"we could not download the artifact ${c.asString} to compute the hash"))
+          case Nil => resolverMonad.raiseError(new RuntimeException(s"we could not download the artifact ${c.asString} to compute the hash for digest type ${digestType} with error ${errorMessage.get}"))
           case head :: tail =>
-            resolverMonad.handleErrorWith(computeSha1(head))(_ => computeSha1s(tail))
+            resolverMonad.handleErrorWith(computeSha(digestType, head)){e => computeShas(digestType, tail, Some(e.getMessage))}
         }
 
+      def processArtifact(src: Artifact.Source, dep: Dependency, proj: Project): Task[Option[JarDescriptor]] = {
+      // TODO, this does not seem like the idea thing, but it seems to work.
+          val artifacts = src.artifacts(dep, proj, None)
+            .iterator
+            .filter(_.url.endsWith(".jar"))
+            .toList
+
+          if (artifacts.isEmpty) Task.point(None)
```

Comment:
can we do a `artifacts match { case Nil => .. ahead :: atail =>` so you can avoid the `.head` call below?